File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        Map<String, Integer> m = new HashMap<>();
        for(int i = 0; i < name.length; i++) {
            m.put(name[i], yearning[i]);
        }
        
        int cnt = 0;
        for(String[] sarr : photo) {
            int ans = 0;
            for(String person : sarr) {
                if (m.containsKey(person)) {
                    ans += m.get(person);
                }
            }
            answer[cnt++] = ans;
        }
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    
    for(vector<string> x : photo) {
        int ans = 0;
        for(string p : x) {
            for(int i = 0; i < name.size(); i++) {
                if (p == name[i]) {
                    ans += yearning[i];
                    break;
                }
            }
        }
        answer.push_back(ans);
    }
    return answer;
}